src/PeopleContext.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { createContext, useEffect, useState } from 'react';
+import { useSearchParams, SetURLSearchParams } from 'react-router-dom';
+import { getPeople } from './api';
+import { Person } from './types';
+
+type Context = {
+  people: Person[]
+  findPersonFather: (people: Person[], person: Person) => string | undefined
+  findPersonMother: (people: Person[], person: Person) => string | undefined
+  isLoading: boolean,
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>
+  setIsPageActive: React.Dispatch<React.SetStateAction<boolean>>
+  error: string | null
+  sex: string
+  centuriesUrl: string[]
+  searchParams: URLSearchParams
+  query: string
+  filterPeople: () => Person[]
+  setSearchParams: SetURLSearchParams
+};
+
+type Props = {
+  children: React.ReactNode
+};
+
+const initialTabContext: Context = {
+  people: [],
+  findPersonFather: () => undefined,
+  findPersonMother: () => undefined,
+  isLoading: false,
+  setIsLoading: () => { },
+  setIsPageActive: () => { },
+  error: null,
+  sex: '',
+  centuriesUrl: [],
+  searchParams: new URLSearchParams(),
+  query: '',
+  filterPeople: () => [],
+  setSearchParams: () => {},
+};
+
+const findPersonFather = (people: Person[], person: Person) => {
+  const father = people.find(p => p.name === person.fatherName);
+
+  return father?.slug;
+};
+
+const findPersonMother = (people: Person[], person: Person) => {
+  const mother = people.find(p => p.name === person.motherName);
+
+  return mother?.slug;
+};
+
+const getCentury = (year: number) => {
+  return (
+    Math.ceil(year / 100).toString()
+  );
+};
+
+export const PeopleContext = createContext<Context>(initialTabContext);
+
+export const PeopleContent: React.FC<Props> = ({ children }) => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isPageActive, setIsPageActive] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || '';
+  const centuriesUrl = searchParams.getAll('centuries') || [];
+  const query = searchParams.get('query') || '';
```

Comment:
doesn't make sense to save all these values in the context
just get them using `useSearchParams` hook

That also means you don't need context at all. For example, the `filterPeople` function is needed only in `PeoplePage` component, so it's better to define it there. The same with other values.

Because in result you've got a big complex Context component with all logic